src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,159 @@
+	private InstanceProvider remoteInstanceProvider;
+
+	private ChainedList openOrdersList;
+	private ChainedList pendingOrdersList;
+	private ChainedList failedOrdersList;
+	private ChainedList spawningOrdersList;
+
+	private String localMemberId;
+
+	/**
+	 * Attribute that represents the thread sleep time when there is no Orders
+	 * to be processed.
+	 */
+	private Long sleepTime;
+
+	private static final Logger LOGGER = Logger.getLogger(OpenProcessor.class);
+
+	public OpenProcessor(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+			String localMemberId, Properties properties) {
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.localMemberId = localMemberId;
+
+		// TODO: ChainedLists instanciation by Singleton pattern.
+
+		String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SLEEP_TIME_KEY,
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.openOrdersList.getNext();
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping OpenProcessor Thread...");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				LOGGER.error("Error while trying to Process an Open Order with the OpenProcessor Thread", e);
+			}
+		}
+	}
+
+	/**
+	 * Method that try to get an Instance for an Open Order. This method can
+	 * generate a race condition. For example: a user can delete a Open Order
+	 * while this method is trying to get an Instance for this Order.
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
+
+				try {
+					InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+					LOGGER.info("Processing Order [" + order.getId() + "]");
+					order.handleOpenOrder();
+					OrderInstance orderInstance = instanceProvider.requestInstance(order);
+					order.setOrderInstance(orderInstance);
+
+					LOGGER.info("Removing Order [" + order.getId() + "] from Open Orders List");
+					this.openOrdersList.removeItem(order);
+
+					LOGGER.info("Updating Order State after processing [" + order.getId() + "]");
+					this.updateOrderStateAfterProcessing(order);
+					
+				} catch (Exception e) {
+					LOGGER.error("Error while trying to get an Instance for Order: " + System.lineSeparator() + order,
+							e);
+					order.setOrderState(OrderState.FAILED);
+
+					LOGGER.info("Adding Order [" + order.getId() + "] to Failed Orders List");
+					this.failedOrdersList.addItem(order);
+				}
+			}
+		}
+	}
+
+	/**
```

Comment:
In general, we prefer to use a more standard javadoc tone "It does something ..."

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/5f772ed160a195f022947266413996241dc24d0f).